### ISSUE TITLE
Decimal Fields not supported?

### ISSUE THREAD

**Issue by maverickandgoose83** [carfup/PCFControls#20]
Any reason these fail? Would have thought they would work since Money is supported?

> An error occurred during the update.
> 
> An error occurred while validating input parameters: Microsoft.OData.ODataException: Cannot convert a value to target type 'Edm.Decimal' because of conflict between input format string/number and parameter 'IEEE754Compatible' false/true.
>    at Microsoft.OData.JsonLight.ODataJsonLightPropertyAndValueDeserializer.ReadPrimitiveValue(Boolean insideJsonObjectValue, IEdmPrimitiveTypeReference expectedValueTypeReference, Boolean validateNullValue, String propertyName)
>    at Microsoft.OData.JsonLight.ODataJsonLightPropertyAndValueDeserializer.ReadNonEntityValueImplementation(String payloadTypeName, IEdmTypeReference expectedTypeReference, PropertyAndAnnotationCollector propertyAndAnnotationCollector, CollectionWithoutExpectedTypeValidator collectionValidator, Boolean validateNullValue, Boolean isTopLevelPropertyValue, Boolean insideResourceValue, String propertyName, Nullable`1 isDynamicProperty)
>    at Microsoft.OData.JsonLight.ODataJsonLightResourceDeserializer.ReadEntryDataProperty(IODataJsonLightReaderResourceState resourceState, IEdmProperty edmProperty, String propertyTypeName)
>    at Microsoft.OData.JsonLight.ODataJsonLightResourceDeserializer.ReadPropertyWithValue(IODataJsonLightReaderResourceState resourceState, String propertyName, Boolean isDeltaResourceSet)
>    at Microsoft.OData.JsonLight.ODataJsonLightResourceDeserializer.<>c__DisplayClass9_0.<ReadResourceContent>b__0(PropertyParsingResult propertyParsingResult, String propertyName)
>    at Microsoft.OData.JsonLight.ODataJsonLightDeserializer.ProcessProperty(PropertyAndAnnotationCollector propertyAndAnnotationCollector, Func`2 readPropertyAnnotationValue, Action`2 handleProperty)
>    at Microsoft.OData.JsonLight.ODataJsonLightResourceDeserializer.ReadResourceContent(IODataJsonLightReaderResourceState resourceState)
>    at Microsoft.OData.JsonLight.ODataJsonLightReader.StartReadingResource()
>    at Microsoft.OData.JsonLight.ODataJsonLightReader.ReadResourceSetItemStart(PropertyAndAnnotationCollector propertyAndAnnotationCollector, SelectedPropertiesNode selectedProperties)
>    at Microsoft.OData.JsonLight.ODataJsonLightReader.ReadAtStartImplementationSynchronously(PropertyAndAnnotationCollector propertyAndAnnotationCollector)
>    at Microsoft.OData.ODataReaderCore.ReadImplementation()
>    at Microsoft.OData.ODataReaderCore.InterceptException[T](Func`1 action)
>    at System.Web.OData.Formatter.Deserialization.ODataReaderExtensions.ReadResourceOrResourceSet(ODataReader reader)
>    at System.Web.OData.Formatter.Deserialization.ODataResourceDeserializer.Read(ODataMessageReader messageReader, Type type, ODataDeserializerContext readContext)
>    at System.Web.OData.Formatter.ODataMediaTypeFormatter.ReadFromStream(Type type, Stream readStream, HttpContent content, IFormatterLogger formatterLogger)
>

**Comment by maverickandgoose83** [comment]
surprised this worked, but just adding this in the retrieveFieldOptions function appears to work

```
case 'decimal':
					let decimalOptions = {
						width : this._context.mode.allocatedWidth,
						label : label,
						fieldDefinition : {
							idDirty : false,
							fieldName : techFieldName,
							fieldType : type,
							controlId : controlId,
							fieldValue : this._parentRecordDetails.Attributes[techFieldName] ?? ""
						},
						disabled : isReadOnly,
						//icon : "Money",
						onClickResult : (fieldDefinition?: DataFieldDefinition) => {
							if(dataFieldDefinitionsDetails != undefined && fieldDefinition != undefined){
								dataFieldDefinitionsDetails.isDirty = true;
								dataFieldDefinitionsDetails.fieldValue = Number(fieldDefinition.fieldValue);

								this._dataFieldDefinitions[this._dataFieldDefinitions.indexOf(dataFieldDefinitionsDetails)] = dataFieldDefinitionsDetails;

								_this._buttonsComponnent.setState({disabled : false});
							}							
						}
					};

					dataFieldDefinitionsDetails = this.completeDataDefinition(dataFieldDefinitionsDetails, decimalOptions.fieldDefinition);

					ReactDOM.render(React.createElement(TextFieldControl, decimalOptions), item);
				break;
```

**Comment by maverickandgoose83** [comment]
looking in the source again, it looks like this code change didn't make it into the latest version?

**Comment by carfup** [comment]
Hello @maverickandgoose83,
I'm sorry here, I didn't get notification on your previous message and since the ticket was closed the same day it was open, I didn't see it at all...
I will include this in the next release for sure.

Thanks for your help here !

**Comment by carfup** [comment]
The fix is now pushed to the release branch : #53
